internal/tmpdir/tmpdir.go

```diff
@@ -18,7 +20,10 @@ const builtinUnixTempDirForBigFiles = "/var/tmp"
 // TemporaryDirectoryForBigFiles returns a directory for temporary (big) files.
 // On non Windows systems it avoids the use of os.TempDir(), because the default temporary directory usually falls under /tmp
 // which on systemd based systems could be the unsuitable tmpfs filesystem.
-func TemporaryDirectoryForBigFiles() string {
+func TemporaryDirectoryForBigFiles(sys *types.SystemContext) string {
+	if sys != nil && sys.BigFilesTemporaryDir != "" {
+		unixTempDirForBigFiles = sys.BigFilesTemporaryDir
```

Comment:
- This still overwrites a global variable; it shouldn’t.
- Why not use the value on Windows as well?

i.e. in here, just `return sys.BigFilesTemporaryDir`.